src/www/status_dhcp_leases.php

```diff
@@ -242,9 +237,13 @@ function remove_duplicate($array, $field)
         }
     }
 
-    if ($_GET['order']) {
-        usort($leases, "leasecmp");
-    }
+    $order = ( $_GET['order'] ) ? $_GET['order'] : 'ip';
+    usort($leases, 
+        function ($a, $b) use ($order) {
+            return strnatcasecmp($a[$order].$a[ip], $b[$order].$b[ip]);
```

Comment:
@ljm42 you seem to miss some quotes here ip -> 'ip', I'm not 100% sure we should concat the ip like this either as it might mangle the requested sort order due to the variable length of the fields. a solution could be to use sprintf() to fix the size of the requested column and concat the address (in the same command) then.

Reply:
Oh right, I'll add those quotes.

The concatenation only affects the comparison when the first term is the same, for instance comparing "static192.168.10.87" to "static192.168.10.101". If the first term is different, the concatenation doesn't matter since the comparison will already have been determined.

Reply:
@ljm42 as long as the field length is the same for the first field, the sorting is fine, when it differs it may lead to unexpected results.

Reply:
It doesn't matter though. If you are comparing "aardvark192.168.10.87" to "zebra192.168.10.101", the one that starts with "aardvark" is always going to sort before the one that starts with "zebra", whether or not an IP address is appended. Changing the second option to "zebra<space><space><space>192.168.10.101" will not change the outcome of the comparison.

The only time the IP address has any effect on the sort value is when the first terms are exactly the same. And then it makes a huge impact on the usability of the page, since you have a nicely sorted list instead of a random one.

(hmm.... I suppose if you were comparing "aardvark192.168.10.87" to "aardvarktwo192.168.10.101", then the IP address would have an effect. But this is highly unlikely given the dataset we're working with, and the unexpected result from that one comparison is far less confusing than the completely unsorted listed we have today)

I encourage you to give this change a try, it is a night and day difference in terms of finding a specific IP on the page.

Reply:
the traditional way, since you already have a custom compare function, is to compare individual sections one after the other and return their compare value if not zero otherwise keep comparing :)

Reply:
Thanks. How's it look now?

Reply:
looks good, thanks!